model_bakery/baker.py

```diff
@@ -357,7 +357,12 @@ def _make(
                 if field.name not in self.model_attrs:
                     self.m2m_dict[field.name] = self.m2m_value(field)
                 else:
-                    self.m2m_dict[field.name] = self.model_attrs.pop(field.name)
+                    if field.name in self.iterator_attrs:
+                        self.model_attrs[field.name] = [
```

Comment:
I'm still a bit unfamiliar with the internals of model-bakery, but when I use `self.m2m_dict` here instead of `self.model_attrs` I keep hitting `StopIteration` for my iterator tests

Here it seems that passing the single-object list with `self.model_attrs` does correctly add the object to the M2M if the model instances in the iterator have been saved

If the model instances in the iterator have not been saved (i.e. `test_create_many_to_many_with_unsaved_iter()`) it seems to hit `_handle_one_to_many()`, which is why I copied the `if not value.pk: value.save()` logic over to there